src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -793,6 +793,22 @@ static void WindowNewRidePaint(rct_window* w, rct_drawpixelinfo* dpi)
     }
 }
 
+static ImageId GetRideImage(RideSelection rideSelection)
+{
+    auto rideEntry = get_ride_entry(rideSelection.EntryIndex);
+    if (rideEntry != nullptr)
+    {
+        for (uint32_t i = 0; i < RCT2::ObjectLimits::MaxRideTypesPerRideEntry; i++)
+        {
+            if (rideEntry->ride_type[i] == rideSelection.Type)
+            {
+                return ImageId(rideEntry->images_offset + i);
+            }
+        }
+    }
+    return ImageId();
+}
```

Comment:
```suggestion
static ImageId GetRideImage(RideSelection rideSelection)
{
    const auto* rideEntry = get_ride_entry(rideSelection.EntryIndex);
    if (rideEntry == nullptr)
        return ImageId{};

    const auto it = std::find(std::begin(rideEntry->ride_type), std::end(rideEntry->ride_type), rideSelection.Type);
    if (it == std::end(rideEntry->ride_type))
        return ImageId{};

    const auto i = std::distance(it, std::end(rideEntry->ride_type));
    return ImageId(rideEntry->images_offset + i);
}
```
In my personal opinion having early exits everywhere makes code a lot easier to read as it reduces scopes which reduces overall line length. Also always use standard algorithms if possible. The code may not compile as I wrote this here on GitHub.